### ISSUE TITLE
[PNI] Search bar becomes hidden on mobile when scrolling down then up

### ISSUE THREAD

**Issue by kristinashu** [mozilla/foundation.mozilla.org#10051]
Issue discovered in - [[PNI] Using backspace in mobile search is very delayed #9821 ](https://github.com/mozilla/foundation.mozilla.org/issues/9821#issuecomment-1399096152)

Via @tbrlpld 
> I am experiencing some weirdness with the visibility of the search bar now. Seems like the search bar is sometimes hidden or blocked by the dropdown when scrolling back up. Also, it sometimes takes quite some time for the grid items to show again. 

This bug has been confirmed by me and @beccaklam

### AC
- [ ] (on mobile) As a user, if I click on the magnifying glass icon to activate search mode, I can see the search bar becomes sticky as I scroll the page.
- [ ] (on mobile) As a user, if search mode is not on, I can see the category bar becomes sticky as I scroll the page.
- [ ] (on mobile) As a user, when search mode is activated, the top right magnifying glass icon becomes a 'x' button which can be used to close the search bar

### Describe the bug
On mobile, the search bar is sometimes hidden or blocked by the dropdown when scrolling down and then back up. 

### To Reproduce
1. Go to https://foundation.mozilla.org/en/privacynotincluded/ with an iPhone
2. Click on the search icon
3. Type something, e.g. "Garmin"
4. Scroll down 
5. Scroll back up
6. See error

### Expected behavior
The active search is still visible when I scroll back up.

### Screenshots
https://user-images.githubusercontent.com/24797493/213828314-353b449b-d1f8-4012-ae6e-e01d8fe9a479.MP4

### Smartphone
 - Device: iPhone12 mini
 - OS: iOS 16.1.2
 - Browser: Safari

**Comment by tbrlpld** [comment]
Thanks for creating this @kristinashu 🙏

**Comment by mmmavis** [comment]
@kristinashu @beccaklam  I'm wondering which component we want to make sticky as user scroll down the page? Right now the behaviour seems a bit unpredictable (buggy)? From the video you attached, the first time you scrolled down, the category nav wasn't sticky and it later became sticky after you scrolled up and down again. Since the purple search box takes over where the category nav is after user clicks on search icon, I'm wondering if we want to change the sticky pattern to

- if purple search box is visible, search box should become sticky (and not the category nav) as users scroll down the page
- if purple search box is not visible (search not toggled on), category nav should be sticky

**Comment by mmmavis** [comment]
Also wondering if we are okay with making the purple search box slightly shorter? Currently the category nav is 52px tall and purple search box is 56px. We can make the search box 52px tall by reducing the padding above and below the input box.

**Comment by beccaklam** [comment]
@mmmavis I like those suggestions! Originally we just had the category nav sticky but I like the idea of making the search bar sticky when there is a search term in place. I'm also fine with reducing the height of the search box to 52px.

**Comment by mmmavis** [comment]
@beccaklam awesome! Just to clarify - are you okay with search bar being sticky if search mode has been toggled on (**with or without a search term**). It gets a bit complicated code-wise if we wanna make search bar sticky ONLY if search term is in place 🙈

**Comment by mmmavis** [comment]
@beccaklam this is on [staging](https://foundation.mofostaging.net/privacynotincluded) for review. I've also updated AC to reflect what we have discussed above.

**Comment by beccaklam** [comment]
Hi @mmmavis as per our slack conversation this needs some additional work before pushing to production. I've added the new AC requirement and am attaching the 'x' svg below along with a [figma link](https://www.figma.com/file/4QFUwtwPN88j74no526S3y/Privacy-Not-Included-Master-File?node-id=3%3A4210&t=WKkZAN3RvzKiX4mL-1) and screen cap. The 'x' svg should be sized at 16px. Thank you!

**'X' SVG:**
![Search-Close](https://user-images.githubusercontent.com/20977376/216471404-90a6da53-33e4-4a23-89cf-b3041a5cc67e.svg)

**Screen cap:**
![image](https://user-images.githubusercontent.com/20977376/216471452-a4b122d6-cb77-49b3-a39a-0c8d053d34f4.png)

**Comment by mmmavis** [comment]
@beccaklam @fessehaye 

I have filed a [PR](https://github.com/mozilla/foundation.mozilla.org/pull/10108) to address the new AC. @fessehaye if PR needs code changes, can you update it for me? This new PR needs to go with [my last PR](https://github.com/mozilla/foundation.mozilla.org/pull/10053) (already in `main`) for next prod push. If the new PR can't make it then we'll have to revert my last PR. I'm gonna be on PTO so won't be able to address code change requests (if any).

**Comment by fessehaye** [comment]
@kristinashu can you verify this is resolved on staging

**Comment by kristinashu** [comment]
Love this is solution, it's so nice! 

@beccaklam @mmmavis as a follow up ticket, do you think the `x` should also clear the search? I find that when a search term is used and then the field hidden, it isn't clear that products are being filtered. This isn't new to this ticket, it's this way on prod right now.

**Comment by beccaklam** [comment]
Hmm, yes I suppose that makes sense! @mmmavis thoughts?

**Comment by mmmavis** [comment]
@kristinashu @beccaklam yup, makes sense to me!

**Comment by kristinashu** [comment]
Ok, here's a ticket for that https://github.com/mozilla/foundation.mozilla.org/issues/10120
